tasks/components/CastIntentTask.js

```diff
@@ -0,0 +1,74 @@
+import h from 'react-hyperscript'
+import { isNil, merge, keys, takeLast, toString, mapObjIndexed, split, forEach, pick, find, equals, values } from 'ramda'
+import { reduxForm as connectForm, Field } from 'redux-form'
+import { compose } from 'recompose'
+
+import ProductList from '../../ordering/components/ProductList'
+import SingleViewProduct from '../../ordering/components/SingleViewProduct'
+
+
+// import styles from '../styles/CastIntentTask'
+
+function CastIntentTask (props) {
+  const { actions, currentAgent, taskPlan, routerParams } = props
+  if(isNil(currentAgent)) {
+    return null
+  }
+
+  const { productId } = routerParams
+  if (productId) {
+    return h(SingleProduct, props)
+  } else {
+    return h(ManyProducts, props)
+  }
+}
+
+function SingleProduct (props) {
+  const { actions, products, routerParams, currentAgent, taskPlan, orderIntents } = props
+  const product = products[routerParams.productId]
+  const nextProps = merge(props, { product, onSubmit })
+  // TODO pass onNavigate to allow SingleViewProduct
+  // to navigate back to list product view
+  return h(SingleViewProduct, nextProps)
+  function onSubmit (value) {
+
+    const submittedOrderIntents = values(mapObjIndexed((quantity, priceSpecString) => {
+      return {
+        agentId: currentAgent.id,
+        desiredQuantity: parseInt(quantity),
```

Comment:
by the way, this number should stay a string: https://github.com/root-systems/cobuy#numbers